### ISSUE TITLE
Names not registered in dictionary() object

### ISSUE THREAD

**Issue by chmue** [kbenoit/quanteda#102]
```r
dct <- dictionary(list(key1 = "word1", key2 = "word2"))
str(dct)
```
```
Formal class 'dictionary' [package "quanteda"] with 1 slot
  ..@ .Data:List of 2
  .. ..$ : chr "word1"
  .. ..$ : chr "word2"
```
```r
dput(dct)
```
```
new("dictionary", .Data = list("word1", "word2"))
```

**Comment by chmue** [comment]
A possible solution would be to have the `dictionary` class derive from `namedList` instead of `list`.

I am currently working on a solution based on this which also fixes #101 in the `dict-fix` branch of my fork.

**Comment by kbenoit** [comment]
It might look a bit strange if you are unfamiliar with the internals of S4 objects, but the names are there:

```
> dct <- dictionary(list(key1 = "word1", key2 = "word2"))
> names(oct)
[1] "key1" "key2"
```
